File: src/main/java/akostenko/aicars/menu/TrackMenu.java
package akostenko.aicars.menu;

import akostenko.aicars.track.StraightTrack;
import akostenko.aicars.track.Track;
import akostenko.aicars.track.CircularTrack;

import java.util.Arrays;
import java.util.List;

public class TrackMenu extends AbstractSubMenu<Track> {

    private final List<Track> items = Arrays.asList(
            new StraightTrack(),
            new CircularTrack()
    );

    @Override
    public String getTitle() {
        return "Track";
    }

    @Override
    public void enter() {
    }

    @Override
    protected List<Track> items() {
        return items;
    }
}

File: src/main/java/akostenko/aicars/menu/TrackMenu.kt
package akostenko.aicars.menu

import akostenko.aicars.track.IndySpeedway
import akostenko.aicars.track.MonzaTrack
import akostenko.aicars.track.Track

class TrackMenu : AbstractSubMenu<Track>() {

    override val items = listOf(MonzaTrack(), IndySpeedway())

    override val title: String = "Track"

    override fun enter() {}
}
